From f414d940d0ea1a8f35ca6d105285336362dff562 Mon Sep 17 00:00:00 2001
From: Marek Vasut <marex@denx.de>
Date: Sat, 20 May 2023 21:43:13 +0200
Subject: [PATCH 4/9] bayer2rgb: Fold src_stride into gst_bayer2rgb_process()

The source stride parameter can be easily obtained from GstBayer2RGB
structure, do it within gst_bayer2rgb_process() and drop the parameter.
No functional change.

Part-of: <https://gitlab.freedesktop.org/gstreamer/gstreamer/-/merge_requests/4686>
---
 gst/bayer/gstbayer2rgb.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/gst/bayer/gstbayer2rgb.c b/gst/bayer/gstbayer2rgb.c
index 07d3d2b..6f1540d 100644
--- a/gst/bayer/gstbayer2rgb.c
+++ b/gst/bayer/gstbayer2rgb.c
@@ -386,8 +386,9 @@ typedef void (*process_func) (guint8 * d0, const guint8 * s0, const guint8 * s1,
 
 static void
 gst_bayer2rgb_process (GstBayer2RGB * bayer2rgb, uint8_t * dest,
-    int dest_stride, uint8_t * src, int src_stride)
+    int dest_stride, uint8_t * src)
 {
+  const int src_stride = GST_ROUND_UP_4 (bayer2rgb->width);
   int j;
   guint8 *tmp;
   process_func merge[2] = { NULL, NULL };
@@ -472,8 +473,7 @@ gst_bayer2rgb_transform (GstBaseTransform * base, GstBuffer * inbuf,
   }
 
   output = GST_VIDEO_FRAME_PLANE_DATA (&frame, 0);
-  gst_bayer2rgb_process (filter, output, frame.info.stride[0],
-      map.data, GST_ROUND_UP_4 (filter->width));
+  gst_bayer2rgb_process (filter, output, frame.info.stride[0], map.data);
 
   gst_video_frame_unmap (&frame);
   gst_buffer_unmap (inbuf, &map);
-- 
2.44.1

